reverse_engineering/helpers/restApiHelper.js

```diff
@@ -48,66 +61,37 @@ class CouchbaseRestApiService {
 		}
 	}
 
-	async count(bucketName) {
-		const body = await this.fetch(`/pools/default/buckets/${bucketName}`);
-
-		return body?.basicStats?.itemCount;
-	}
-
-	async getDocuments(bucketName, size) {
-		let docsUri = `/pools/default/buckets/${bucketName}/docs?include_docs=true&limit=${size}&skip=0`;
-
-		const body = await this.fetch(docsUri);
-
-		return body?.rows?.map(item => {
-			return {
-				json: safeParse(item?.doc?.json),
-				meta: {
-					id: item?.id,
-				},
-			};
-		});
-	}
-
-	async getDocumentByKey(bucketName, key) {
-		let docsUri = `/pools/default/buckets/${bucketName}/docs/${encodeURIComponent(key.replace(/\u0000/g, ''))}`;
-		const body = await this.fetch(docsUri);
-
-		return {
-			...body,
-			json: safeParse(body.json),
-		};
-	}
-
-	async getRandomKey(bucketName) {
-		const randomKeyUri = `/pools/default/buckets/${bucketName}/localRandomKey`;
-
-		return this.fetch(randomKeyUri);
+	/**
+	 * @param {{ bucketName: string; scopeName: string; collectionName: string; limit: number }} param0
+	 * @returns {Promise<Document[]>}
+	 */
+	async getCollectionDocuments({ bucketName, scopeName, collectionName, limit }) {
+		const endpoint = `/pools/default/buckets/${bucketName}/scopes/${scopeName}/collections/${collectionName}/docs?include_docs=true&limit=${limit}&skip=0`;
+		return await this.fetch(endpoint);
 	}
 
-	async getVersion() {
-		const body = await this.fetch('/pools');
-
-		return body.implementationVersion;
+	/**
+	 * @param {{ bucketName: string; scopeName: string; collectionName: string; }} param0
```

Comment:
JSDoc annotations are great.
It would be nice to have a short description of the functins too.